File: 19-Iterator/src/go/song.go
package main

import "fmt"

// Song is the element stored in the collection.
type Song struct {
	title  string
	artist string
}

func NewSong(title, artist string) Song {
	return Song{title: title, artist: artist}
}

func (s Song) String() string {
	return fmt.Sprintf("%q by %s", s.title, s.artist)
}

File: 19-Iterator/src/rust/song.rs
use std::fmt;

/// The element stored in the collection.
#[derive(Clone)]
pub struct Song {
    title: String,
    artist: String,
}

impl Song {
    pub fn new(title: &str, artist: &str) -> Self {
        Song { title: title.to_string(), artist: artist.to_string() }
    }
}

impl fmt::Display for Song {
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        write!(f, "\"{}\" by {}", self.title, self.artist)
    }
}
